modules/waypoint_names_to_coordinates.py

```diff
@@ -0,0 +1,31 @@
+"""
+This file contains a function for converting waypoint names to coordinates.
+"""
+
+def waypoint_names_to_coordinates(waypoint_names: "list[str]",
+                waypoint_mapping: "dict[str, tuple[float, float]]") \
+        -> "tuple[bool, list[tuple[float, float]]]":
+    """
+    Converts a list of waypoint names to their corresponding coordinates based on a waypoint mapping.
+
+    Args:
+        waypoint_names (list[str]): A list of waypoint names.
+        waypoint_mapping (dict[str, tuple[float, float]]): A dictionary mapping waypoint names to their coordinates.
+
+    Returns:
+        tuple[bool, list[tuple[float, float]]]: A tuple containing a boolean indicating the success of the conversion
+            and a list of coordinates corresponding to the given waypoint names.
+    """
+    coordinates = []
+
+    if not len(waypoint_names) == 0:  # Handle the case when waypoint_names is empty
```

Comment:
len() gets the count of objects in the container. This conditional has non-empty containers go to line 22, which doesn't seem correct...